cinema/models.py

```diff
@@ -84,23 +86,47 @@ class Ticket(models.Model):
     row = models.IntegerField()
     seat = models.IntegerField()
 
-    def clean(self):
-        for ticket_attr_value, ticket_attr_name, cinema_hall_attr_name in [
-            (self.row, "row", "rows"),
-            (self.seat, "seat", "seats_in_row"),
-        ]:
-            count_attrs = getattr(
-                self.movie_session.cinema_hall, cinema_hall_attr_name
+    class Meta:
+        constraints = [
+            UniqueConstraint(fields=["row", "seat", "movie_session"],
+                             name="unique_ticket")
+        ]
+
+    @staticmethod
+    def validate_seat(
+            row: int,
+            seat: int,
+            num_rows: int,
+            num_seats: int,
+            error: callable):
```

Comment:
The `validate_seat` method's `error` parameter should be documented to clarify its expected type and usage. This will help avoid misuse and make the code more understandable.